### PR TITLE
Adopt glTF convention: lights face local neg-z direction

### PR DESCRIPTION
**Pull request by WestLangley**
As discussed in https://github.com/mrdoob/three.js/pull/14775#issuecomment-421525219.

For now, only `RectAreaLight` is governed by its quaternion, so it is the only light for which this new convention applies.

This change required a rewrite of `RectAreaLightHelper`. The helper must now be added as a child of the light, which is probably a better API anyway.

### CODE REVIEW

**Comment by donmccurdy** [comment]
glTF does not currently define area lights, so this isn't a conformance concern. However — do we think similar changes to other directed lights (Spot, Directional) will eventually be wanted? I.e. removing `light.target` in favor of local direction affected by the node's world rotation? If that's the end goal, then +1 for updating RectAreaLight and using -Z for lights going forward. If we think the other directed lights will remain as-is, then I have no preference on this PR.

**Comment by WestLangley** [comment]
>However — do we think similar changes to other directed lights (Spot, Directional) will eventually be wanted? I.e. removing light.target in favor of local direction affected by the node's world rotation?

I believe so.

**Comment by mbredif** [comment]
+1 for having the same convention for lights and cameras (lookat faces neg-z instead of pos-z for lights and neg-z for cameras), as it simplifies shadowmapping with this shared convention (proposed in https://github.com/mrdoob/three.js/pull/14658#issuecomment-413037953 already).

File: src/core/Object3D.js [hunk 1]
```diff
@@ -332,7 +332,7 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 			position.setFromMatrixPosition( this.matrixWorld );
 
-			if ( this.isCamera ) {
+			if ( this.isCamera || this.isLight ) {
```
Comment (mrdoob) [inline src/core/Object3D.js]: This change is scary, but less see what happens...

**Comment by mrdoob** [comment]
Thanks!

File: src/core/Object3D.js (hunk 1 above)
Comment (WestLangley) [reply src/core/Object3D.js]: We'll see... Thanks for giving this PR a go!

**Pull request merged**